src/Net/Http.fs

```diff
@@ -554,20 +554,39 @@ module private HttpHelpers =
         hasContentType.Value
 
     let getResponse (req:HttpWebRequest) silentHttpErrors =
-        if defaultArg silentHttpErrors false then
+
+        let getResponseAsync (req:HttpWebRequest) =
             async {
-                try
+                #if FX_NO_WEBREQUEST_TIMEOUT
                     return! Async.FromBeginEnd(req.BeginGetResponse, req.EndGetResponse)
-                with
-                    | :? WebException as exn -> 
-                        if exn.Response <> null then 
-                           return exn.Response
-                        else 
-                            reraisePreserveStackTrace exn
-                            return Unchecked.defaultof<_>
+                #else
+                    let! child = 
+                        Async.FromBeginEnd(req.BeginGetResponse, req.EndGetResponse)
+                        |> fun x -> Async.StartChild(x, req.Timeout)
```

Comment:
Could you please split this into two `let` bindings? I know people who prefer piping into lambda functions, but I find it confusing :-). I means something like:

    let op = Async.FromBeginEnd(...)
    let! child = Async.StartChild(...)